### PR TITLE
LPS-162136 Adjust ObjectDefinition DTO and use ObjectFieldSetting

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issues: 
https://issues.liferay.com/browse/LPS-162136;
https://issues.liferay.com/browse/LPS-162853.

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:objects

**Comment by p-albuquerque** [comment]
ci:test:object

**Comment by MarinhoFeliphe** [comment]
taking a look

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 1]
```diff
@@ -345,6 +345,9 @@ private ObjectDefinition _toObjectDefinition(
 				pluralLabel = LocalizedMapUtil.getLanguageIdMap(
 					objectDefinition.getPluralLabelMap());
 				portlet = objectDefinition.getPortlet();
+				restContextPath =
+					objectDefinition.isSystem() ? StringPool.BLANK :
```
[the remark this answers is not in the document]
Comment (MarinhoFeliphe) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java]: I believe we can get the proper context path from `SystemObjectDefinitionMetadata`

File: modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java [hunk 2]
```diff
@@ -72,6 +74,53 @@ public void tearDown() throws Exception {
 		}
 	}
 
+	@Override
+	@Test
+	public void testGetObjectDefinitionsPage() throws Exception {
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java:80]: Override this one `testGetObjectDefinition` as we talked about.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 3]
```diff
@@ -301,6 +303,16 @@ private void _testAddObjectRelationship(String type) throws Exception {
 				objectFieldNamePrefix +
 					_objectDefinition1.getPKObjectFieldName()));
 
+		ObjectFieldSetting objectFieldSetting =
+			_objectFieldSettingLocalService.fetchObjectFieldSetting(
+				objectRelationship.getObjectFieldId2(),
+				"objectDefinitionName1");
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:316]: Define the name in `ObjectFieldSettingConstants`

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:stop

**Comment by p-albuquerque** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java (hunk 3 above)
Comment (p-albuquerque) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java]: next commit fix this

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java [hunk 4]
```diff
@@ -72,6 +74,35 @@ public void tearDown() throws Exception {
...
+		if (objectDefinition.getSystem()) {
+			Assert.assertTrue(
+				StringUtil.startsWith(
+					objectDefinition.getRestContextPath(), ""));
+		}
+		else {
+			Assert.assertTrue(
+				StringUtil.startsWith(
+					objectDefinition.getRestContextPath(), "/o/c/"));
+		}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java:98]:
````
Assert.assertEquals(StringBundler.concat(
  objectDefinition.getSystem() ? "/o/"  : "/o/c/",
  TextFormatter.formatPlural(objectDefinition.getName())),
 objectDefinition.getRestContextPath());
````

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 5]
```diff
@@ -353,9 +354,16 @@ private ObjectDefinition _toObjectDefinition(
...
+					if (StringUtil.startsWith(
+							objectDefinition.getRESTContextPath(), "/")) {
+
+						restContextPath =
+							"/o" + objectDefinition.getRESTContextPath();
+					}
+					else {
+						restContextPath =
+							"/o/" + objectDefinition.getRESTContextPath();
+					}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:366]:
````
restContextPath = StringBundler.concat(
objectDefinition.isSystem() ? "/o/" : "/o", objectDefinition.getRESTContextPath());
````

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java (hunk 5 above)
Comment (p-albuquerque) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:366]: formatsource don't let me do exactly this,
so I do:
`				restContextPath = objectDefinition.isSystem() ? "/o/" :
					"/o" + objectDefinition.getRESTContextPath();`

**Comment by p-albuquerque** [comment]
ci:close

**Pull request closed without merging**